source/influxdb/api.d

```diff
@@ -731,22 +724,70 @@ private bool valueIsFloat(in string value) @safe pure nothrow @nogc {
     m.to!string.shouldEqualLine(`cpu\ "load"\,\ test,tag\ 1=to"to,tag\,2=foo foo\,\=\ ="a,b",b\,a\=r="a \" b"`);
 }
 
+enum ValueType { bool_, int_, float_, string }
+
 struct InfluxValue {
 
-    string value;
+    import std.typecons : Nullable;
 
-    this(T)(T v) if(is(T == long) || is(T == int)) {
-        import std.conv: to;
-        value = v.to!string ~ "i";
+    union Payload {
+        bool b;
+        long i;
+        double f;
     }
 
-    this(T)(T v) if(is(T == string) || is(T == double) || is(T == bool)) {
-        import std.conv: to;
-        value = v.to!string;
+    private {
+        Payload m_value;
```

Comment:
No `m_` prefix please, just `_value`. `_svalue` isn't a good name though, shouldn't it be part of the union instead? As `string rawString` maybe?